CorsixTH/Lua/hospitals/player_hospital.lua

```diff
@@ -47,175 +47,234 @@ function PlayerHospital:PlayerHospital(world, avail_rooms, name)
 end
 
 --! Give advice to the player at the end of a day.
+--! Note: For patient/staff warmth checks, advice is only on one or the other per month
 function PlayerHospital:dailyAdviceChecks()
   local current_date = self.world:date()
   local day = current_date:dayOfMonth()
-  local current_month = current_date:monthOfYear()
 
   -- Hold any advice back until the game has somewhat started.
   if current_date < Date(1, 5) then
     return
   end
 
-  -- Check for advice on money.
-  -- This must occur after monthly maintenance and salary costs
-  -- or it may give invalid advice
-  if day == 1 then
-    if not self.world.free_build_mode then
-      if self.balance < 2000 and self.balance >= -500 then
-        local cashlow_advice = {
-          _A.warnings.money_low, _A.warnings.money_very_low_take_loan,
-          _A.warnings.cash_low_consider_loan,
-        }
-        self:giveAdvice(cashlow_advice)
-
-      elseif self.balance < -2000 and current_month > 8 then
-        -- TODO: Ideally this should be linked to the lose criteria for balance.
-        self:giveAdvice({_A.warnings.bankruptcy_imminent})
-
-      elseif self.balance > 6000 and self.loan > 0 then
-        self:giveAdvice({_A.warnings.pay_back_loan})
-      end
-    end
-  end
+  -- Track sitting / standing ratio of patients.
+  if day % ratio_interval == 0 then
+    -- Compute the ratio of today.
+    local num_sitting, num_standing = self:countSittingStanding()
+    local ratio = (num_sitting + num_standing > 10)
+        and num_sitting / (num_sitting + num_standing) or nil
 
-  -- Warn about lack of a staff room.
-  if day == 3 and self:countRoomOfType("staff_room", 1) == 0 then
-    local staffroom_advice = {
-      _A.warnings.build_staffroom, _A.warnings.need_staffroom,
-      _A.warnings.staff_overworked, _A.warnings.staff_tired,
-    }
-    self:giveAdvice(staffroom_advice)
+    -- Store the measured ratio.
+    -- Once we hit num_sitting_ratios of index points, we wrap around
+    self.adviser_data.sitting_ratios[self.adviser_data.sitting_index] = ratio
+    self.adviser_data.sitting_index = (self.adviser_data.sitting_index >= num_sitting_ratios)
+        and 1 or self.adviser_data.sitting_index + 1
   end
 
-  -- Warn about lack of toilets.
-  if day == 8 and self:countRoomOfType("toilets", 1) == 0 then
-    local toilet_advice = {
-      _A.warnings.need_toilets, _A.warnings.build_toilets,
-      _A.warnings.build_toilet_now,
-    }
-    self:giveAdvice(toilet_advice)
+  if day == 1 then
+    self:_adviseMoney()
+  end
+  if day == 3 then
+    self:_adviseStaffRoom()
   end
-
   -- Make players more aware of the need for radiators
-  if self:countRadiators() == 0 then
+  if day == 5 and self:countRadiators() == 0 then
     self:giveAdvice({_A.information.initial_general_advice.place_radiators})
   end
+  if day == 8 then
+    self:_adviseToilets()
+  end
```

Comment:
cool. This reads much better. `elseif` probably isn't important except as a microoptimization but I would have expected it.